src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+ * by specified device
+ */
+static constexpr Key<std::map<ie::Precision, float>, PropertyMutability::RO> gops{"DEVICE_GOPS"};
+
+/**
+ * @brief  Read only property to get a float of device thermal
+ */
+static constexpr Key<float, PropertyMutability::RO> thermal{"DEVICE_THERMAL"};
+
+}  // namespace device
+
+/**
+ * @brief The key for enabling of dumping the topology with details of layers and details how
+ * this network would be executed on different devices to the disk in GraphViz format.
+ */
+static constexpr Key<bool, PropertyMutability::RW> dump_graph_dot{"HETERO_DUMP_GRAPH_DOT"};
+
+/**
+ * @brief The key with the list of device targets used to fallback unsupported layers
+ * by HETERO plugin
+ */
+static constexpr device::Priorities target_fallback{"TARGET_FALLBACK"};
+
+namespace execution {
+
+/**
+ * @brief The number of executor logical partitions
+ */
+static constexpr Key<int32_t, PropertyMutability::RW> streams{"STREAMS"};
+
+/**
+ * @brief Maximum number of concurent tasks executed by execututor
+ */
+static constexpr Key<int32_t, PropertyMutability::RW> concurrency{"CONCURRENCY"};
+
+/**
+ * @brief Enum to define possible affinity patterns
+ */
+enum class Affinity {
+    NO = -1,   //!<  Disable threads affinity pinning
+    CORE = 0,  //!<  Pin threads to cores, best for static benchmarks
+    NUMA = 1,  //!<  Pin threads to NUMA nodes, best for real-life, contented cases. On the Windows and MacOS* this
+               //!<  option behaves as CORE
+    HYBRID_AWARE = 2,  //!< Let the runtime to do pinning to the cores types, e.g. prefer the "big" cores for latency
+                       //!< tasks. On the hybrid CPUs this option is default
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Affinity& affinity) {
+    switch (affinity) {
+    case Affinity::NO:
+        return os << "NO";
+    case Affinity::CORE:
+        return os << "CORE";
+    case Affinity::NUMA:
+        return os << "NUMA";
+    case Affinity::HYBRID_AWARE:
+        return os << "HYBRID_AWARE";
+    default:
+        throw ov::Exception{"Unsupported affinity pattern"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, Affinity& affinity) {
+    std::string str;
+    is >> str;
+    if (str == "NO") {
+        affinity = Affinity::NO;
+    } else if (str == "CORE") {
+        affinity = Affinity::CORE;
+    } else if (str == "NUMA") {
+        affinity = Affinity::NUMA;
+    } else if (str == "HYBRID_AWARE") {
+        affinity = Affinity::HYBRID_AWARE;
+    } else {
+        throw ov::Exception{"Unsupported affinity pattern: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief The name for setting CPU affinity per thread option.
+ * @note The setting is ignored, if the OpenVINO compiled with OpenMP and any affinity-related OpenMP's
+ * environment variable is set (as affinity is configured explicitly)
+ */
+static constexpr Key<Affinity> affinity{"AFFINITY"};
```

Comment:
was `CPU_BIND_THREAD`, is it compatible?

Reply:
yes https://github.com/openvinotoolkit/openvino/pull/9587